airflow/providers/common/sql/hooks/sql.py

```diff
@@ -181,19 +182,21 @@ def __init__(self, *args, schema: str | None = None, log_sql: bool = True, **kwa
             "replace_statement_format", "REPLACE INTO {} {} VALUES ({})"
         )
 
-    @property
+    @cached_property
     def placeholder(self):
-        conn = self.get_connection(getattr(self, self.conn_name_attr))
+        conn_id = getattr(self, self.conn_name_attr)
```

Comment:
One small thing that I noticed - is that for some reason few lines below the `conn_name_attr` is casted to str. I guess it was from some past where mypy has been complaining - maybe we can remove the cast there as well - since it seems it does not complain any more.

Reply:
I don't understand what you mean by casting,  here I just keep the result of that getattr so I can also print it in the warning message if needed.  I think this getattr thing is a generic thing as some hooks have different attribute names to define the connection id.  Maybe something to cleanup when going to Airflow 3.0. and generalize it across all hooks.

Reply:
my comment that I just accidentally noticed tha tfew lines below (just scroll down) below your change the same snippet uses `cast` - which likely might be removed as part of this change - because apparently mypy does not complain any more here (Apparently someone added cast in the past because it did complain for some reason).

Reply:
```python
    def get_conn(self):
        """Return a connection object."""
        db = self.get_connection(getattr(self, cast(str, self.conn_name_attr))) # <--- PROBABLY WE CAN REMOVE THIS CAST
        return self.connector.connect(host=db.host, port=db.port, username=db.login, schema=db.schema)
```

Reply:
Ow @potiuk now I get it :)

Reply:
Will try to remove that ;)

Reply:
> ```python
>     def get_conn(self):
>         """Return a connection object."""
>         db = self.get_connection(getattr(self, cast(str, self.conn_name_attr))) # <--- PROBABLY WE CAN REMOVE THIS CAST
>         return self.connector.connect(host=db.host, port=db.port, username=db.login, schema=db.schema)
> ```

I even went a bit further, I replaced all duplicated code regarding this into a property to keep code DRY, I think there were 4 or 5 duplications in the DbApiHook regarding this while one was string casted like you mentioned and the others werent:

```
@property
def conn_id(self) -> str:
    return getattr(self, self.conn_name_attr)
```

Reply:
Nice!